Source/Urho3D/CSharp/Swig/strings.i

```diff
@@ -13,19 +13,88 @@
 #include <EASTL/string.h>
 %}
 
-namespace eastl {
+%wrapper %{
+    SWIGEXPORT void SWIGSTDCALL CSharp_Urho3D_String_Set(eastl::string* str, const char* val) { *str = val; }
+    SWIGEXPORT const char* SWIGSTDCALL CSharp_EaStl_String_Get(eastl::string* str) { return str->data(); }
```

Comment:
data() or c_str()?

Reply:
It is same thing, but typemaps are more universal with `data()`, because `string_view` does not have `c_str()`.

Reply:
data() doesn't have trailing error. See skipped unit test for Buffer in c#

Reply:
Not sure what you mean. `data()` and `c_str()` were always identical. Before c++11 `data()` had no requirement to be zero-terminated, but due to practical reasons it always was. Since c++11 this is now a requirement for `data()` as well. All of this is for strings of course. `string_view` may have strings not terminated with zero and that is ok, this case is handled. I also could not find any "buffer" test in our C# tests.